internal/factsengine/mapper.go

```diff
@@ -130,7 +130,11 @@ func FactsGatheredToEvent(gatheredFacts entities.FactsGathered) ([]byte, error)
 		FactsGathered: facts,
 	}
 
-	eventBytes, err := events.ToEvent(&event, entities.FactsGathererdEventSource, uuid.New().String())
+	eventBytes, err := events.ToEvent(
+		&event,
+		events.WithSource(entities.FactsGathererdEventSource),
+		events.WithID(uuid.New().String()),
```

Comment:
If we don't use the `WithID`, the uuid will be generated in the contract side in the same way.

Reply:
Yes it's true, I wanted just to point it out as a matter of clarity, anyway we can discuss further on the sensible details of the contracts :D